### ISSUE TITLE
ofxNetwork : wrongly reported errors

### ISSUE THREAD

**Issue by oxillo** [openframeworks/openFrameworks#6457]
ofxNetwork may report errors that should be ignored.
This happens when there are construct like : 

```cpp
int err= ofxNetworkCheckError();
if ( err != OFXNETWORK_ERROR(NOTSOCK) ){
	// Ignore this error
}
``` 
As ofxNetworkCheckError() gets the error and logs it, all the errors will be reported even if they should be **ignored**.

This explains the error seen here:
```
[  error ] ofxNetwork: C:/projects/openframeworks/addons/ofxNetwork/src/ofxTCPManager.cpp: 200 ENOTSOCK: socket argument is not a socket
```

https://ci.appveyor.com/project/arturoc/openframeworks/builds/28847653/job/0duai87vuou8ctv7#L1358

_Originally posted by @ofTheo in https://github.com/openframeworks/openFrameworks/issues/6412#issuecomment-553731166_

## Proposal
 - Deprecate ofxNetworkCheckError()
- Add `int ofxNetworkGetLastError()` to return the last error
- Add void ofxNetworkLogError(int err, const char* file, int line) to log an error that happened on `line` line in file `file`
- Add ofxNetworkLogLastError() that is a shortcut for `ofxNetworkLogError(ofxNetworkGetLastError(), __FILE__, __LINE__)`

**Comment by P-v-K** [comment]
Should that read "all the errors will be _**ignored**_ even if they should be reported."?

**Comment by oxillo** [comment]
It's more "all the errors will be reported even if they should be **ignored**". I'll edit the post to make it clearer.

**Comment by ofTheo** [comment]
@oxillo so would ofxNetworkGetLastError() return no error for errors that can be safely ignored? 
By the way I have also seen this issue occasionally on the CI for macOS makefile tests.

**Comment by ofTheo** [comment]
This is one of the last issues left before we can release 0.11.1. 
I get the sense that it could be quite a lot of changes. Do you think we should push this to 0.11.2 / 0.12 ?

**Comment by P-v-K** [comment]
I have no opinion on pushing or fixing, as I just ran into it once and it hasn't been impacting my current work. Maybe it could be mentioned in the documentation to help people be more likely to figure out what's (not) happening if/when they do run into it? Because once you know, and find the lines of code mentioned, you could hack it to do what you want? Just a thought, trying to be helpful. Again, I don't have enough knowledge/experience with it to have a strong opinion.

**Comment by oxillo** [comment]
The idea behind my proposal is to separate getting the error and logging it.
So you can decide to print or not the error. 
See/Comment the above PR

**Comment by ofTheo** [comment]
closed by #6675
